senaite-openmrs/src/main/java/com/ozonehis/eip/openmrs/senaite/mapper/senaite/AnalysisRequestMapper.java

```diff
@@ -0,0 +1,92 @@
+import org.springframework.stereotype.Component;
+
+@Setter
+@Component
+public class AnalysisRequestMapper {
+
+    private static final Logger log = LoggerFactory.getLogger(AnalysisRequestMapper.class);
+
+    public AnalysisRequest toSenaite(
+            Contact contact, AnalysisRequestTemplate analysisRequestTemplate, ServiceRequest serviceRequest) {
+        if (serviceRequest == null) {
+            return null;
+        }
+        serviceRequest.getCode().getCodingFirstRep().getCode();
+
+        AnalysisRequest analysisRequest = new AnalysisRequest();
+
+        analysisRequest.setContact(contact.getUid());
+        analysisRequest.setDateSampled(convertOpenmrsDateToSenaiteDate(
+                serviceRequest.getOccurrencePeriod().getStart()));
+        analysisRequest.setClientSampleID(serviceRequest.getIdPart());
+        analysisRequest.setReviewState("sample_due");
+
+        if (analysisRequestTemplate.getAnalysisRequestTemplateItems() != null
+                && !analysisRequestTemplate.getAnalysisRequestTemplateItems().isEmpty()) {
+
+            analysisRequest.setTemplate(analysisRequestTemplate
+                    .getAnalysisRequestTemplateItems()
+                    .get(0)
+                    .getUid());
+
+            if (analysisRequestTemplate.getAnalysisRequestTemplateItems().get(0).getSampleType() != null) {
+                analysisRequest.setSampleType(analysisRequestTemplate
+                        .getAnalysisRequestTemplateItems()
+                        .get(0)
+                        .getSampleType()
+                        .getUid());
+            }
+            if (analysisRequestTemplate.getAnalysisRequestTemplateItems().get(0).getAnalysisProfile() != null) {
+                analysisRequest.setProfiles(analysisRequestTemplate
+                        .getAnalysisRequestTemplateItems()
+                        .get(0)
+                        .getAnalysisProfile()
+                        .getUid());
+            }
+            if (analysisRequestTemplate.getAnalysisRequestTemplateItems().get(0).getAnalyses() != null) {
+                analysisRequest.setAnalyses(getAnalysesUids(analysisRequestTemplate
+                        .getAnalysisRequestTemplateItems()
+                        .get(0)
+                        .getAnalyses()));
+            }
+        }
+
+        return analysisRequest;
+    }
+
+    private String[] getAnalysesUids(Analyses[] analysesList) {
+        String[] uids = new String[analysesList.length];
+        for (int i = 0; i < analysesList.length; i++) {
+            uids[i] = analysesList[i].getServiceUid();
+        }
+        return uids;
+    }
+
+    public String convertOpenmrsDateToSenaiteDate(Date openmrsDate) {
+        long timestampInSeconds = openmrsDate.toInstant().getEpochSecond();
+        Instant instant = Instant.ofEpochSecond(timestampInSeconds);
+
+        return DateTimeFormatter.ISO_OFFSET_DATE_TIME.withZone(ZoneOffset.UTC).format(instant);
```

Comment:
Do you need to format it in UTC? Is this a requirement? Otherwise I believe generated string will contain the server timezone.

Reply:
Okay. 
Replaced with 
```
DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(instant);
```